conda/gateways/disk/read.py

```diff
@@ -58,23 +61,30 @@ def yield_lines(path):
             raise
 
 
-def _digest_path(algo, path):
-    if not isfile(path):
+@deprecated("23.9", "24.3", addendum="Use `conda.gateways.disk.read.compute_sum` instead.")
+def _digest_path(algo: Literal["md5", "sha256"], path: str | os.PathLike) -> str:
+    return compute_sum(path, algo)
+
+
+# FUTURE: Python 3.11+, replace with hashlib.file_digest
+def compute_sum(path: str | os.PathLike, algo: Literal["md5", "sha256"]) -> str:
```

Comment:
I'm not sure if we need to check for the file existence here anymore, we could already mimic the `hashlib.filedigest` API here to make the deprecation easier long-term. IOW I'd rather have the core of this be a backport of `file_digest` that we remove at some point.

```suggestion
# FUTURE: Python 3.11+, replace with hashlib.file_digest
def file_digest(fileobj, digest: Literal["md5", "sha256"], /) -> str:
```